src/generator/WDL/entities/WorkflowGenerator.js

```diff
@@ -62,46 +86,97 @@ export default class WorkflowGenerator {
     return '';
   }
 
-  buildCallMapping(children) {
+  genCall(child) {
     const EOL = this.settings.getValue('style.eol');
     const SCOPE_INDENT = this.settings.getValue('style.scope_indent');
     const DOUBLE_SCOPE_INDENT = `${SCOPE_INDENT}${SCOPE_INDENT}`;
     const TRIPLE_SCOPE_INDENT = `${DOUBLE_SCOPE_INDENT}${SCOPE_INDENT}`;
 
     let res = '';
 
-    _.forEach(children, (val) => {
-      let callString = `${SCOPE_INDENT}${constants.CALL} `;
+    const val = child;
+    let callString = `${SCOPE_INDENT}${constants.CALL} `;
 
-      if (val.action.name === val.name) {
-        callString += `${val.name}`;
-      } else {
-        callString += `${val.action.name} ${constants.AS} ${val.name}`;
-      }
+    if (val.action.name === val.name) {
+      callString += `${val.name}`;
+    } else {
+      callString += `${val.action.name} ${constants.AS} ${val.name}`;
+    }
 
-      res += `${callString}`;
+    res += `${callString}`;
 
-      if (_.size(val.i) > 0) {
-        let mappingWasFounded = false;
+    if (_.size(val.i) > 0) {
+      let mappingWasFounded = false;
 
-        _.forEach(val.i, (pVal, pKey) => {
-          if (pVal.inputs && _.size(pVal.inputs) > 0) {
-            if (!mappingWasFounded) {
-              res += ` ${constants.SCOPE_OPEN}${EOL}`;
-              res += `${DOUBLE_SCOPE_INDENT}${constants.INPUT}${constants.COLON}${EOL}`;
-              mappingWasFounded = true;
-            }
-            res += `${TRIPLE_SCOPE_INDENT}${pKey} ${constants.EQ} ${this.buildPortValue(pVal)},${EOL}`;
+      _.forEach(val.i, (pVal, pKey) => {
+        if (pVal.inputs && _.size(pVal.inputs) > 0) {
+          if (!mappingWasFounded) {
+            res += ` ${constants.SCOPE_OPEN}${EOL}`;
+            res += `${DOUBLE_SCOPE_INDENT}${constants.INPUT}${constants.COLON}${EOL}`;
+            mappingWasFounded = true;
           }
-        });
-        if (mappingWasFounded) {
-          res += `${SCOPE_INDENT}${constants.SCOPE_CLOSE}`;
+          res += `${TRIPLE_SCOPE_INDENT}${pKey} ${constants.EQ} ${this.buildPortValue(pVal)},${EOL}`;
         }
+      });
+      if (mappingWasFounded) {
+        res += `${SCOPE_INDENT}${constants.SCOPE_CLOSE}`;
       }
+    }
+
+    return `${res}${EOL}`;
+  }
```

Comment:
Thank you so it is a one more great thing done by your review!

It is special case for call generating function because it could be leaded by different way (it could have the "input mapping block" or could not)  and so it is impossible to place ${EOL} by way you recommened.

Let me leave it if you do not mindly?